workspaces/src/types/block.rs

```diff
@@ -32,18 +37,42 @@ impl Block {
 
 /// The block header info. This is a non-exhaustive list of items that
 /// could be present in a block header. More can be added in the future.
-#[derive(Clone, Debug, PartialEq)]
-struct BlockHeader {
-    height: BlockHeight,
-    epoch_id: CryptoHash,
-    hash: CryptoHash,
-    timestamp_nanosec: u64,
+///
+/// NOTE: For maintainability purposes, some items have been excluded. If required,
+/// please submit an issue to [workspaces](https://github.com/near/workspaces-rs/issues).
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub struct BlockHeader {
+    pub height: BlockHeight,
+    pub epoch_id: CryptoHash,
+    pub next_epoch_id: CryptoHash,
+    pub hash: CryptoHash,
+    pub prev_hash: CryptoHash,
+    pub timestamp_nanosec: u64,
+    pub random_value: CryptoHash,
+    pub gas_price: Balance,
+    pub block_ordinal: Option<u64>,
+    pub total_supply: Balance,
+    pub last_final_block: CryptoHash,
+    pub last_ds_final_block: CryptoHash,
+    pub next_bp_hash: CryptoHash,
+    pub latest_protocol_version: u32,
+
+    pub prev_state_root: CryptoHash,
+    pub chunk_receipts_root: CryptoHash,
+    pub chunk_headers_root: CryptoHash,
+    pub chunk_tx_root: CryptoHash,
+    pub outcome_root: CryptoHash,
+    pub challenges_root: CryptoHash,
+    pub block_merkle_root: CryptoHash,
```

Comment:
Instead of making these fields public, I suggest we expose getters. That way it will be easier to rename/remove/add fields when we need it, without breaking changes.